src/oca_github_bot/tasks/rebase_bot.py

```diff
@@ -0,0 +1,92 @@
+from ..manifest import _user_can_push
+from ..process import CalledProcessError, check_call
+from ..queue import getLogger, task
+
+_logger = getLogger(__name__)
+
+
+@task()
+@switchable("rebase_bot")
+def rebase_bot_start(org, repo, pr, username, dry_run=False):
+    with github.login() as gh:
+        gh_pr = gh.pull_request(org, repo, pr)
+        target_branch = gh_pr.base.ref
+        try:
+            with github.temporary_clone(org, repo, target_branch) as clone_dir:
+                if not _user_can_push(
+                    gh, org, repo, username, clone_dir, target_branch
+                ):
+                    github.gh_call(
+                        gh_pr.create_comment,
+                        f"Sorry @{username} you are not allowed to rebase.\n\n"
+                        f"To do so you must either have push permissions on "
+                        f"the repository, or be a declared maintainer of all "
+                        f"modified addons.\n\n"
+                        f"If you wish to adopt an addon and become it's "
+                        f"[maintainer]"
+                        f"(https://odoo-community.org/page/maintainer-role), "
+                        f"open a pull request to add "
+                        f"your GitHub login to the `maintainers` key of its "
+                        f"manifest.",
+                    )
+                    return
+                # get remote name
+                remote = gh_pr.head._repo_owner
+                if not remote:
+                    github.gh_call(
+                        gh_pr.create_comment,
+                        f"Sorry @{username}, impossible to rebase "
+                        f"because unknown remote.",
+                    )
+                    return
+                # fetch pr and create local branch
+                current_branch = github.git_get_current_branch(cwd=clone_dir)
+                local_branch = f"{remote}_{current_branch}_{pr}"
+                check_call(
+                    ["git", "fetch", "origin", f"refs/pull/{pr}/head:{local_branch}"],
+                    cwd=clone_dir,
+                )
+                check_call(["git", "checkout", current_branch], cwd=clone_dir)
+                # do rebase
+                check_call(["git", "fetch", "origin", target_branch], cwd=clone_dir)
+                check_call(
+                    ["git", "rebase", "--autosquash", target_branch], cwd=clone_dir
+                )
+                # push rebase
+                check_call(
+                    [
+                        "git",
+                        "push",
+                        "--force",
+                        f"https://github.com/{remote}/{repo}.git",
+                        f"{local_branch}:{current_branch}",
+                    ],
+                    cwd=clone_dir,
+                    log_error=False,
+                )
+                check_call(["git", "branch", "-D", f"{local_branch}"], cwd=clone_dir)
```

Comment:
Don't bother deleting the local branch, since we are in a temporary clone directory that will deleted automatically.